mu-plugins/blocks/time/src/index.js

```diff
@@ -0,0 +1,83 @@
+/**
+ * WordPress dependencies
+ */
+import { useSelect } from '@wordpress/data';
+import { RichTextToolbarButton } from '@wordpress/block-editor';
+import { useCallback } from '@wordpress/element';
+import { getTextContent, registerFormatType, slice, toggleFormat } from '@wordpress/rich-text';
+
+/**
+ * External dependencies
+ */
+import { gmdate, strtotime } from 'locutus/php/datetime';
+
+/**
+ * Internal dependencies
+ */
+import metadata from './block.json';
+
+const { name, icon, title } = metadata;
+
+const Edit = ( { isActive, onChange, value } ) => {
+	const { date_gmt } = useSelect( ( select ) => select( 'core/editor' ).getCurrentPost() );
+
+	const toggleWithoutEnhancing = useCallback( () => {
+		onChange(
+			toggleFormat( value, {
+				type: name,
+			} )
+		);
+	} );
+
+	return (
+		<RichTextToolbarButton
+			icon={ icon }
+			title={ title }
+			onClick={ () => {
+				const dateDescription = getTextContent( slice( value ) );
+
+				if ( ! dateDescription || isActive ) {
+					toggleWithoutEnhancing();
+
+					return;
+				}
+
+				// Remove the word "at" from the string, if present.
+				// Allows strings like "Monday, April 6 at 19:00 UTC" to work.
+				const dateCleaned = dateDescription.replace( 'at ', '' );
```

Comment:
(I was curious about why this was here - looks like it's from the original shortcode)